src/gfx/pal_spec.cpp

```diff
@@ -203,26 +203,36 @@ static T readLE(U const *bytes) {
 
 /*
  * **Appends** the first line read from `file` to the end of the provided `buffer`.
+ *
+ * @return true if a line was read.
  */
-static void readLine(std::filebuf &file, std::string &buffer) {
+[[gnu::warn_unused_result]] // Ignoring EOF is a bad idea.
+static bool readLine(std::filebuf &file, std::string &buffer) {
 	// TODO: maybe this can be optimized to bulk reads?
 	for (;;) {
 		auto c = file.sbumpc();
 		if (c == std::filebuf::traits_type::eof()) {
-			return;
+			return false;
 		}
 		if (c == '\n') {
 			// Discard a trailing CRLF
 			if (!buffer.empty() && buffer.back() == '\r') {
 				buffer.pop_back();
 			}
-			return;
+			return true;
 		}
 
 		buffer.push_back(c);
 	}
 }
 
+#define requireLine(kind, file, buffer) do { \
+	if (!readLine(file, buffer)) { \
+		error(kind " palette file appears to be truncated"); \
```

Comment:
How about `kind " palette file ends too early"`? (It definitely does, and this is common error-message phrasing for a file that, well, ends too early.)

Reply:
It's more technically precise, but I think it will confuse a non-technical user more. “File is truncated”, I believe, will mean more to them—as slightly less vague than “File is corrupted”.

Reply:
Interesting, I was thinking "ends too early" sounded *less* technical than "truncated".

Reply:
RGBFIX and RGBGFX both use "file too short" when their input binary files don't have enough bytes; how about that?

Reply:
Yes; actually, your previous comment made me think of exactly that. Let's.